analyzers/tests/SonarAnalyzer.UnitTest/TestCases/DisablingCSRFProtection.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace TestCases
+{
+    public class DisablingCSRFProtection
+    {
+        public static void ConfigureServices(IServiceCollection services)
+        {
+            services.AddControllersWithViews(options => options.Filters.Add(new IgnoreAntiforgeryTokenAttribute())); // Noncompliant
+            services.AddControllersWithViews(options => options.Filters.Add(new AutoValidateAntiforgeryTokenAttribute()));
+            services.AddControllersWithViews(options => options.Filters.Add(new ValidateAntiForgeryTokenAttribute()));
+            services.AddControllers(options => options.Filters.Add(new IgnoreAntiforgeryTokenAttribute())); // Noncompliant
+            services.AddMvc(options => options.Filters.Add(new IgnoreAntiforgeryTokenAttribute())); // Noncompliant
+            services.AddMvcCore(options => options.Filters.Add(new IgnoreAntiforgeryTokenAttribute())); // Noncompliant
+
+            Action<MvcOptions> setupAction = options => options.Filters.Add(new IgnoreAntiforgeryTokenAttribute()); // Noncompliant
+            services.AddControllersWithViews(setupAction);
+
+            static void SetupAction(MvcOptions options) => options.Filters.Add(new IgnoreAntiforgeryTokenAttribute()); // Noncompliant
+            services.AddControllersWithViews(SetupAction);
+
+            services.AddMvc(options => options.Filters.Add(GetAntiForgeryPolicy()));
+
+            IgnoreAntiforgeryTokenAttribute _ = new(); // Noncompliant
+        }
+
+        private static IAntiforgeryPolicy GetAntiForgeryPolicy() =>
+            new IgnoreAntiforgeryTokenAttribute(); // Noncompliant
+    }
+
+    [IgnoreAntiforgeryToken] // Noncompliant
+    public class S4502Controller : Controller
+    {
+        private readonly string path = string.Empty;
+
+        [HttpPost]
+        [IgnoreAntiforgeryToken] // Noncompliant
+        public IActionResult ChangeEmail_Noncompliant(string model) => View(path);
+
+        [HttpPost, IgnoreAntiforgeryTokenAttribute] // Noncompliant
```

Comment:
can you also add a test with `[HttpGet, IgnoreAntiforgeryTokenAttribute, AnotherAttribute]` ?

and one with only `[IgnoreAntiforgeryToken]` ?

is it related only to HttpPost?

Reply:
> can you also add a test with [HttpGet, IgnoreAntiforgeryTokenAttribute, AnotherAttribute] ?
> and one with only [IgnoreAntiforgeryToken] ?

As far as I can tell we don't need these cases. We register for `AttributeSyntax` and we will get triggered for each occurrence.

> is it related only to HttpPost?

No, it's independent. I've added [HttpPost] just to cover the case when the attribute is in a list and not alone like on the line 40.

Reply:
I was asking for these from a black-box testing point of view, from a functional point of view. It's good to have edge-case scenarios.

The edge cases I meant were:
- a scenario without "HttpPost" on the method. If I come after one year, I may be tempted to change the implementation to check if "HttpPost" is present on the method, and tests will be green. From the current documentation (i.e. tests), I see that this rule should only raise on "HttpPost" methods.
-  a scenario where "IgnoreAntiforgeryTokenAttribute" is not the last attribute in a list of attributes. Maybe that's exaggerated, but I try to cover edge cases. And I don't see major downsides.

The implementation can change. This is a simple rule, but in general, implementation can change and existing test cases should ensure a good enough safety net for refactoring.

Reply:
I usually prefer a red-green-refactor approach in order to be able to add only the tests and code we need.

As discussed offline I've removed the `[HttpPost]` attribute from line 39 to have a case with just an attribute.